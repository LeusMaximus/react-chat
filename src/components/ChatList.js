// React
import React from 'react';

// MUI components
import List from '@material-ui/core/List';

// vendor modules
import nanoid from 'nanoid';

// Own modules
import ChatListItem from './ChatListItem';

const ChatList = ({ chats }) => (
  <List>
    {chats.map(item => <ChatListItem key={nanoid()} item={item} />)}
  </List>
);

export default ChatList;
