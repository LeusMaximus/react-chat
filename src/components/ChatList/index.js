// React
import React from 'react';
import PropTypes from 'prop-types';

// MUI components
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

// vendor modules
import nanoid from 'nanoid';

// Own modules
import ChatListItem from '../ChatListItem';
import { IChatItem } from '../../interfaces/propTypes';

const ChatList = ({
  chats, setActiveChat, activeId, searchTerm, disabled,
}) => {
  const filteredChats = chats
    .filter(item => item.title && item.title.toLowerCase().includes(searchTerm.toLowerCase()))
    .sort((one, two) =>
      (one.title && two.title && one.title.toLowerCase() <= two.title.toLowerCase() ? -1 : 1));

  return (
    <List component="div">
      {filteredChats.length ? (
        filteredChats.map(item => (
          <ChatListItem
            key={nanoid()}
            item={item}
            setActiveChat={setActiveChat}
            activeId={activeId}
            disabled={disabled}
          />
        ))
      ) : (
        <ListItem component="div">
          <ListItemText primary="No chats found..." />
        </ListItem>
      )}
    </List>
  );
};

ChatList.propTypes = {
  chats: PropTypes.arrayOf(IChatItem.isRequired).isRequired,
  setActiveChat: PropTypes.func.isRequired,
  activeId: PropTypes.string.isRequired,
  searchTerm: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
};

export default ChatList;
