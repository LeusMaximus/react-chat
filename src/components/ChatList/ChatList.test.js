/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ChatList from './index';

jest.mock('../ChatListItem', () => ({ item }) => item.title);

const mockProps = {
  chats: [
    {
      _id: '1000',
      title: 'Chat 1',
      createdAt: '2014-02-16T10:44:32.200Z',
      updatedAt: '2017-04-14T11:45:35.200Z',
    },
    {
      _id: '2000',
      title: 'Chat 2',
      createdAt: '2017-04-12T11:24:35.200Z',
      updatedAt: '2017-04-14T11:45:35.200Z',
    },
  ],
  disabled: false,
  setActiveChat: jest.fn(),
  activeId: '1000',
  searchTerm: '',
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot test with chats', () => {
    const tree = renderer.create(<ChatList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('snapshot test without chats', () => {
    const tree = renderer.create(<ChatList {...mockProps} chats={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
