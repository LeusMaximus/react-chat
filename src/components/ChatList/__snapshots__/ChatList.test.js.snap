// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> snapshot test with chats 1`] = `
<div
  className="MuiList-root-5 MuiList-padding-6"
>
  Chat 1
  Chat 2
</div>
`;

exports[`<ChatList /> snapshot test without chats 1`] = `
<div
  className="MuiList-root-5 MuiList-padding-6"
>
  <div
    className="MuiListItem-root-9 MuiListItem-default-12 MuiListItem-gutters-16"
    disabled={false}
  >
    <div
      className="MuiListItemText-root-19"
    >
      <span
        className="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
      >
        No chats found...
      </span>
    </div>
  </div>
</div>
`;
